RFCs/Image Filter.md

```diff
@@ -0,0 +1,156 @@
+Propose an additional feature or set of features that enable users to query kpack images based on their `STATUS` and `REASON` such that it is easier for users to perform the workflows in the “Use Cases” section.
+
+Although the feature request stated in the “Problem” section is to query builds, it is probably more productive to query images because end-users most likely care about the current state of their application build and not builds that happened in the past.  Additionally, querying based on builds would pose a technical problem as the CLI would need to present builds and their build numbers outside the context of their associated image.  
+
+### Prior Art:
+
+* Build Reason RFC - Addresses the problem of bringing clarity to why another build was triggered
+    * This RFC is trying to solve the somewhat related problem of providing clarity to the user after an action is performed that causes rebuilds/rebases
+    * https://github.com/sampeinado/kpack/blob/master/rfcs/0002-reason-message-diff.md
+* User interview with Jason Morgan (VMware internal) https://docs.google.com/document/d/11NiLB3LJ0sK3LzmwNga_pFOI939p7lBL7Vq1SKFKAZw/edit
+    * Issue that stemmed from the user interview with Jason so the context is available to the community https://github.com/vmware-tanzu/kpack-cli/issues/79 
+
+### Complexity/Risk
+
+* By introducing a new column in `kp image list`, we may encounter a similar width problem as we did with the builds list table
+
+### Alternatives:
+
+* Users parse their image list as it currently exists and scroll through it until they find the images with the attributes they care about
+* Usage of kubectl to return images with certain attributes
+
+### Mockups
+
+Below I will cover specific scenarios made possible by a filtering feature. All the following scenarios are derived from this proposed modification to the `kp image list` command:
+
+```
+kp image list --status <string> --reason <string1, string2 ...>
+```
+
+Rather than adding an additional subcommand to the `kp image list` path like `filter` or `query`, I propose adding an additional `LATEST REASON` column to the existing output of the `kp image list` table. This would also allow for future design extension. If we wanted to add additional filtering capabilities, we would just create new attribute flags with values as arguments.
+
+Note: The mockups are not meant to provide sample output for all possible scenarios when using this command. Rather they focus on the workflows described in the "Use Cases" section and some non-obvious scenarios that are made possible by introduction of the new command.
+
+Note: `SHA1` is used in the mockups that display images with a `ready` status so that rows don't wrap in the body of the PR
+
+------------------------------
+
+**Sample Help Output**
+
+```
+$ kp image list -h
+
+Prints a table of the most important information about images in the provided namespace.
+
+The namespace defaults to the kubernetes current-context namespace.
```

Comment:
We could add `--all-namespaces` in this rfc if we wanted

Reply:
`--all-namespaces` / `-A` for across all namespaces would make the scope so much bigger. But, that makes me wonder about it's behaviour in a multi tenant environment (like Shipyard)

Reply:
That's a good question. I think similar to running the same command with kubectl you would get an access denied error.